Don't contribute a currency field when in the 'fake' migration world

vanschelven opened the pull request:
in that case the currency field should/will simply be added because it is explicitly defined as part of some migration.

See #725

benjaoming commented:
@vanschelven thanks so much for this impressive work! I think that your explanation makes sense and the change looks like the right one.

But I would just defer to @Stranger6667 for a comment here?

Would you mind adding a release note entry referencing the findings and explanation in #725 ? The bug and its fix is important for others to be aware of, as both the bug and the subsequent fix cause migrations to happen certain circumstances - right? :pray:

vanschelven commented:
I've added a note to the changelog. Your understanding is correct: because this is now fixed, upgrading to the version of django-money with this fix will start detecting migrations, which may in itself be surprising.

benjaoming reviewed (approved):
Thanks for adding release notes! I don't have anything to add here, will ask @Stranger6667 to have a look :hourglass_flowing_sand:

Stranger6667 reviewed (approved):
Looks good to me!

benjaoming commented:
Great! We'd better bump the release to 3.3 because of the significance of the change...

stianjensen commented:
Great stuff! I'm pretty sure this is what's been blocking us from upgrading from 2.1.1. Will test this to confirm our issue is also gone now :+1:

vanschelven commented:
If we're being "paranoid" (and given the lack of tests) I'd like to point out one more thing.

In [the code that I used as an example of how it used to work when "south" was still a thing](https://github.com/django-money/django-money/commit/69b51755f5bd18e4b85f9887aced69998d462e66#diff-0b1ce35a3277e63b0a1f1bd7c798dbc656d3ab504c188d6d9553095759caf800L283) there is one more if-statement like the one I added in this PR.

Given my analysis on the issue-description that if-statement seems redundant to me, and it did not appear to be necessary in the manual tests that I ran on my live project, but if we want to do "better safe than sorry" we could add it in as well.

stianjensen commented:
> Can you report back if that's the case? I'll put out the final release after the additional verification and encouragement

All good from me on this! We've been using the currency_field_name to share currency between multiple moneyfields and also adding a custom CharField to our models, and some of the hacks we've had on our end are now unnecessary, probably thanks to this fix.

vanschelven commented:
> Great! We'd better bump the release to 3.3 because of the significance of the change...

@benjaoming kind reminder to do this

benjaoming commented:
Released here: https://pypi.org/project/django-money/
